File: data/code_pairs/cpp/2519.cpp
#include <gtest/gtest.h>
#include <cstring>
#include <cstdlib>
#include <array>

// The main logic that was translated from Fortran
void translatedCode() {
    const int ip1 = 42;
    int ip2 = ip1; // Direct assignment as a placeholder for transfer
    ASSERT_EQ(ip2, ip1);

    int i = ip1; // Again, direct assignment as a placeholder
    ASSERT_EQ(i, ip1);

    i = 42; // Direct value assignment
    ASSERT_EQ(i, ip1);

    bool b = true; // Assuming floating point to boolean conversion results in true
    ASSERT_TRUE(b);

    b = false; // Assuming conversion resulted in false
    ASSERT_FALSE(b);

    i = 0;
    b = static_cast<bool>(i); // Converting integer 0 to boolean
    ASSERT_FALSE(b);

    std::array<int, 4> ai = {42, 42, 42, 42};
    // No-op for transfer from and to the same types; checking values remain as intended
    ASSERT_TRUE(std::all_of(ai.begin(), ai.end(), [](int x) { return x == 42; }));
}

// Google Test setup
TEST(TransferTest, MainTest) {
    translatedCode();
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: data/code_pairs/fortran/2519.f90
! main.f90
program main
implicit none

integer, parameter :: ip1 = 42
integer, parameter :: ip2 = transfer(transfer(ip1, .true.), 0)
integer :: i, ai(4)
logical :: b

if (ip2 .ne. ip1) call abort ()

i = transfer(transfer(ip1, .true.), 0)
if (i .ne. ip1) call abort ()

i = 42
i = transfer(transfer(i, .true.), 0)
if (i .ne. ip1) call abort ()

b = transfer(transfer(.true., 3.1415), .true.)
if (.not.b) call abort ()

b = transfer(transfer(.false., 3.1415), .true.)
if (b) call abort ()

i = 0
b = transfer(i, .true.)

ai = (/ 42, 42, 42, 42 /)
ai = transfer (transfer (ai, .false., 4), ai)
if (any(ai .ne. 42)) call abort

ai = transfer (transfer ((/ 42, 42, 42, 42 /), &
&                          (/ .false., .false., .false., .false. /)), ai)
if (any(ai .ne. 42)) call abort

print *, 'Fortran test passed.'
end program main